core/src/main/java/org/infinispan/transaction/impl/AbstractCacheTransaction.java

```diff
@@ -59,7 +61,9 @@
    private final AtomicReference<Set<Object>> lockedKeys = new AtomicReference<>();
 
    /** Holds all the locks for which the local node is a secondary data owner. */
-   private final AtomicReference<Set<Object>> backupKeyLocks = new AtomicReference<>();
+   @GuardedBy("this")
+   private Map<Object, CompletableFuture<Void>> backupKeyLocks;
```

Comment:
Could you modify the comment to say what these CompletableFutures actually represent?